### PR TITLE
CSHARP-3454:Expressions for datetime truncation and binning.

### PR DESCRIPTION
**Pull request by rstam**

### CODE REVIEW

**Comment by rstam** [comment]
Evergreen patch build:

https://evergreen.mongodb.com/version/618c426e2a60ed2ef6a228cc

File: tests/MongoDB.Driver.Tests/Linq/Linq3ImplementationTests/Jira/CSharp3454Tests.cs [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+{
+    public class CSharp3454Tests : Linq3IntegrationTest
+    {
+        [Fact]
+        public void Truncate_with_Day_should_work()
+        {
+            var collection = GetCollection<C>();
+
+            var queryable = collection.AsQueryable()
+                .Select(x => x.DateTime.Truncate(DateTimeUnit.Day));
```
Comment (rstam) [inline tests/MongoDB.Driver.Tests/Linq/Linq3ImplementationTests/Jira/CSharp3454Tests.cs:31]: Each call to `.Select(...)` in this file illustrates (and tests) a different variation of the C# API for `$dateTrunc`.

File: src/MongoDB.Driver/Linq/DateTimeExtensions.cs [hunk 2]
```diff
@@ -278,5 +278,42 @@ public static DateTime Subtract(this DateTime @this, long value, DateTimeUnit un
...
+        }
+
+        /// <summary>
+        /// Truncates a DateTime value to the specified unit and bin size.
+        /// </summary>
+        /// <param name="this">The original DateTime.</param>
+        /// <param name="unit">The unit.</param>
+        /// <param name="binSize">The bin size.</param>
+        /// <returns>The resulting DateTime.</returns>
+        public static DateTime Truncate(this DateTime @this, DateTimeUnit unit, int binSize)
```
Comment (JamesKovacs) [inline src/MongoDB.Driver/Linq/DateTimeExtensions.cs:300]: The server represents `binSize` as a 64-bit integer. Should we declare `binSize` as a long to match the server? I can't imagine someone wanting to bin on more than `int.MaxValue`, but the server does support up to `long.MaxValue`.

**Review by JamesKovacs — APPROVED** [review]
LGTM. Question about whether `binSize` should be an `int` as is currently implemented or changed to a `long` to match the server.

**Comment by alexmurari** [comment]
@rstam There will be a 2.14 beta 2 version of the driver? I'm really looking forward to test the new LINQ3 provider (especially the `$$ROOT` support). Actually, there is a nuget feed for nightly builds?

**Comment by rstam** [comment]
We're not planning a beta2 version of 2.14.0, but 2.14.0 will be released soon.

We don't have a nuget feed for nightly builds.

**Pull request merged**